src/libsync/capabilities.cpp

```diff
@@ -176,6 +176,25 @@ bool Capabilities::chunkingNg() const
     return _capabilities["dav"].toMap()["chunking"].toByteArray() >= "1.0";
 }
 
+PushNotificationTypes Capabilities::pushNotificationsAvailable() const
+{
+    PushNotificationTypes pushNotificationTypes;
+
+    if (!_capabilities.contains("notify_push"))
+        return pushNotificationTypes;
+
+    const auto &types = _capabilities["notify_push"].toMap()["type"].toStringList();
+    if (types.contains("files"))
+        pushNotificationTypes.setFlag(PushNotificationType::Files);
+
+    return pushNotificationTypes;
+}
+
+QString Capabilities::pushNotificationWebSocketUrl() const
```

Comment:
I suggest constructing a QUrl instance here since we are going to do it later anyway and trying to search for **Url() const** in our code-base, I can see that in most cases if the method is said to return a URL, it will return a QUrl.

Reply:
Yes that would make more sense.